homework/index.js

```diff
@@ -29,26 +37,163 @@
+    return firstRepo.name.localeCompare(secondRepo.name);
+  }
+
+  function renderRepositoryNames(repositories, select) {
+    repositories.forEach((repository, index) => {
+      createAndAppend('option', select, {
+        text: repository.name,
+        value: index,
+      });
+    });
+  }
+
+  function sortAndMapRepositories(repositoryInfos) {
+    return repositoryInfos
+      .sort(sortRepositoriesByNameAscending)
+      .map(repositoryInfo => ({
+        name: repositoryInfo.name,
+        url: repositoryInfo.html_url,
+        description: repositoryInfo.description,
+        forks: repositoryInfo.forks,
+        lastUpdateTime: getDateTimeText(repositoryInfo.updated_at),
+        contributorsUrl: repositoryInfo.contributors_url,
+      }));
+  }
+
   function renderRepoDetails(repo, ul) {
-    createAndAppend('li', ul, { text: repo.name });
+    const repoDetailsListItem = createAndAppend('li', ul, {
+      class: 'repository-item',
+    });
+    const repoDetailsTable = createAndAppend('table', repoDetailsListItem, {
+      class: 'repository-item-table',
+    });
+    appendRepoDetail('Repository:', repo.name, repoDetailsTable, repo.url);
+    appendRepoDetail('Description:', repo.description, repoDetailsTable);
+    appendRepoDetail('Forks:', repo.forks, repoDetailsTable);
+    appendRepoDetail('Updated:', repo.lastUpdateTime, repoDetailsTable);
+  }
+
+  function renderContributorDetails(contributor, ul, index) {
+    const repoContributorListItem = createAndAppend('li', ul, {
+      class: 'contributors-item',
+    });
+    if (index === 0) {
+      createAndAppend('h6', repoContributorListItem, {
+        text: 'Contributions:',
+      });
+    }
+    createAndAppend('img', repoContributorListItem, {
+      src: contributor.avatar_url,
+      alt: 'Contributor avatar picture',
+    });
+    createAndAppend('a', repoContributorListItem, {
+      href: contributor.html_url,
+      text: contributor.login,
+      target: '_blank',
+    });
+    createAndAppend('span', repoContributorListItem, {
+      text: contributor.contributions,
+    });
+  }
+
+  function renderHeaderSection(title, parent) {
+    const header = createAndAppend('header', parent);
+    createAndAppend('span', header, {
+      text: title,
+      class: 'header-title',
+    });
+    return createAndAppend('select', header);
+  }
+
+  function renderSubSection(sectionId, parent) {
+    const subSection = createAndAppend('section', parent, {
+      class: sectionId,
+      id: sectionId,
+    });
+    createAndAppend('ul', subSection, { class: 'sub-container' });
+  }
+
+  function renderMainSectionWithSubSections(parent) {
+    const mainSection = createAndAppend('main', parent, { class: 'container' });
+    renderSubSection(REPOSITORY_DETAIL_SECTION_ID, mainSection);
+    renderSubSection(REPOSITORY_CONTRIBUTORS_SECTION_ID, mainSection);
+  }
+
+  function getUnorderedListAsEmpty(sectionId) {
+    const section = document.getElementById(sectionId);
```

Comment:
Here you are creating using `id` attributes to get access to elements that are created elsewhere in your code. You could have saved references to those elements when you created them, rather than placing "flag poles" in the DOM to find them again later with `.getElementById()`. You could either pass references to the relevant elements through parameters to the functions that need them or, alternatively, you could create a global object, and store the references there:

```js
'use strict';

{
  const dom = {};
  ...

function renderSubSection(sectionId, parent) {
  dom[sectionId] = createAndAppend('section', parent, {
    class: sectionId,
  });
  createAndAppend('ul', subSection, { class: 'sub-container' });
}
```

Personally, I would prefer explicitly passing the references using the parameter lists.